client/src/modules/vouchers/voucher-registry.ctrl.js

```diff
@@ -149,28 +134,21 @@ function VoucherController(Vouchers, Notify, uiGridGroupingConstants,
     vm.hasError = false;
     toggleLoadingIndicator();
 
-    Vouchers.read(null, filters).then(function (vouchers) {
-      vm.gridOptions.data = vouchers;
-
-      // loop through the vouchers and precompute the voucher type tags
-      vouchers.forEach(function (voucher) {
-        var transaction = get(voucher.type_id);
-        voucher._isIncome = (transaction.type === INCOME);
-        voucher._isExpense = (transaction.type === EXPENSE);
-        voucher._isOther = (transaction.type === OTHER);
-        voucher._type = transaction.text;
-      });
-
-      vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.ALL);
-    })
-      .catch(function (err) {
-        if (err && !err.code) { return; }
-        Notify.handleError(err);
+    Vouchers.read(null, filters)
+      .then(function (vouchers) {
+        vm.gridOptions.data = vouchers;
+
+        vouchers.forEach(function (voucher) {
+          // determine the transaction_type for this voucher
+          var transactionType = transactionTypeMap[voucher.type_id];
+          voucher._isIncome = (transactionType.type === INCOME);
```

Comment:
In the case where the `voucher.type_id` == null, the `transactionTypeMap[null]` == undefined.

The line 144 : throw an error `Cannot read the property type` because transactionType is undefined.

Reply:
Whoops!  I thought `type_id`s were required.